src/classes/AFFL_ContactChange_TDTM.cls

```diff
@@ -102,23 +105,23 @@ public with sharing class AFFL_ContactChange_TDTM extends TDTM_Runnable {
 
 	                   	//CONTACT FIELD CHANGED
 	                   	if(affl.Primary__c && affl.Contact__c != null && afflOld.Contact__c != null
-	                   		&& affl.Contact__c != afflOld.Contact__c && newRelatedContact != null && oldRelatedContact != null && lookupFieldName != null) {
+	                   		&& affl.Contact__c != afflOld.Contact__c && newRelatedContact != null && oldRelatedContact != null && lookupFieldAPIName != null) {
```

Comment:
some questions here: same as before with the checking for null and then checking if they're aren't equal, if the old contact field is null, and we are changing to not be null, do we need to allow for that?  Secondly, if affOld.Contact is essentially oldReleatedContact, do we need both of those comparisons? I see that one is the contact itself and one is the contact Id, so I think that case could we just use newRelated and oldRelated and drop the other two?

Reply:
I have actually just changed the label name here, I am not entirely sure how the logic works. Please let me know if you want me to dig deeper.

Reply:
well it certainly wouldn't hurt to gain some understanding at a bit lower level, but it's not something I would hold up a merge for.